homeassistant/components/music_assistant/media_player.py

```diff
@@ -0,0 +1,729 @@
+                self.hass, media_id, self.entity_id
+            )
+            media_id = sourced_media.url
+            media_id = async_process_play_media_url(self.hass, media_id)
+
+        if announce:
+            await self._async_play_announcement(
+                media_id,
+                use_pre_announce=kwargs[ATTR_MEDIA_EXTRA].get("use_pre_announce"),
+                announce_volume=kwargs[ATTR_MEDIA_EXTRA].get("announce_volume"),
+            )
+            return
+
+        # forward to our advanced play_media handler
+        await self._async_play_media_advanced(
+            media_id=media_id if isinstance(media_id, list) else [media_id],  # type: ignore[unreachable]
+            enqueue=enqueue,
+            media_type=media_type,
+            radio_mode=kwargs[ATTR_MEDIA_EXTRA].get(ATTR_RADIO_MODE),
+        )
+
+    @catch_musicassistant_error
+    async def async_join_players(self, group_members: list[str]) -> None:
+        """Join `group_members` as a player group with the current player."""
+        player_ids: list[str] = []
+        for child_entity_id in group_members:
+            # resolve HA entity_id to MA player_id
+            if (hass_state := self.hass.states.get(child_entity_id)) is None:
+                continue
+            if (mass_player_id := hass_state.attributes.get("mass_player_id")) is None:
+                continue
+            player_ids.append(mass_player_id)
+        await self.mass.players.cmd_sync_many(self.player_id, player_ids)
+
+    @catch_musicassistant_error
+    async def async_unjoin_player(self) -> None:
+        """Remove this player from any group."""
+        await self.mass.players.player_command_unsync(self.player_id)
+
+    @catch_musicassistant_error
+    async def _async_play_media_advanced(
+        self,
+        media_id: list[str],
+        artist: str | None = None,
+        album: str | None = None,
+        enqueue: MediaPlayerEnqueue | QueueOption | None = None,
+        radio_mode: bool | None = None,
+        media_type: str | None = None,
+    ) -> None:
+        """Send the play_media command to the media player."""
+        # pylint: disable=too-many-arguments
+        media_uris: list[str] = []
+        item: MediaItemType | None = None
+        # work out (all) uri(s) to play
+        for media_id_str in media_id:
+            # URL or URI string
+            if "://" in media_id_str:
+                media_uris.append(media_id_str)
+                continue
+            # try content id as library id
+            if media_type and media_id_str.isnumeric():
+                with suppress(MediaNotFoundError):
+                    if media_type is not None:
+                        continue
+                    item = await self.mass.music.get_item(  # type: ignore[unreachable]
+                        MediaType(media_type), media_id_str, "library"
+                    )
+                    if item.uri is not None:
+                        continue
```

Comment:
why would uri be None? I would assume if its not playable it would raise `MediaNotFoundError`?